kobo/apps/organizations/tests/test_organization_invitations.py

```diff
@@ -63,57 +65,86 @@ def _update_invite(self, user, guid, status):
         self.client.force_login(user)
         return self.client.patch(self.detail_url(guid), data={'status': status})
 
-    def test_owner_can_send_invitation(self):
+    @data(
+        ('owner', status.HTTP_201_CREATED),
+        ('admin', status.HTTP_201_CREATED),
+        ('member', status.HTTP_403_FORBIDDEN),
+        ('external', status.HTTP_403_FORBIDDEN)
+    )
+    @unpack
+    def test_owner_can_send_invitation(self, user_role, expected_status):
         """
-        Test that the organization owner can create invitations
+        Test that only organization owner or admin can create invitations
         """
-        response = self._create_invite(self.owner_user)
-        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
-        self.assertEqual(
-            len(response.data), len(self.invitation_data['invitees'])
-        )
-        for index, invitation in enumerate(response.data):
-            self.assertIn(
-                invitation['invitee'], self.invitation_data['invitees']
-            )
-            # Check that the email was sent
-            invite = User.objects.filter(
-                Q(username=invitation['invitee']) |
-                Q(email=invitation['invitee'])
-            ).first()
+        user = getattr(self, f'{user_role}_user')
+        response = self._create_invite(user)
+        self.assertEqual(response.status_code, expected_status)
+        if response.status_code == status.HTTP_201_CREATED:
             self.assertEqual(
-                mail.outbox[index].to[0],
-                invite.email if invite else invitation['invitee']
+                len(response.data), len(self.invitation_data['invitees'])
             )
+            for index, invitation in enumerate(response.data):
+                self.assertIn(
+                    invitation['invitee'], self.invitation_data['invitees']
+                )
+                # Check that the email was sent
+                invite = User.objects.filter(
+                    Q(username=invitation['invitee']) |
+                    Q(email=invitation['invitee'])
+                ).first()
+                self.assertEqual(
+                    mail.outbox[index].to[0],
+                    invite.email if invite else invitation['invitee']
+                )
 
-    def test_owner_can_resend_invitation(self):
+    @data(
+        ('owner', status.HTTP_200_OK),
+        ('admin', status.HTTP_200_OK),
+        ('member', status.HTTP_403_FORBIDDEN),
+        ('external', status.HTTP_403_FORBIDDEN)
```

Comment:
I think it should 404 for external. We don't want to reveal the existence of the organization to external.